### PR TITLE
arious fixes, some just paranoid

### PR DESCRIPTION
**Pull request by aqualast**
There are still problems with realloc() to be unfixed, whose return value would be nullptr if reallocation failed and should be checked. However there's currently no failure path (error code or exception) in its call sites, so I haven't touched any of them.

### CODE REVIEW

File: wintaser/AVIDumper.cpp [hunk 1]
```diff
@@ -1196,9 +1198,9 @@ bool OpenAVIFile(int width, int height, int bpp, int fps)
 			// BITMAPINFOHEADER
 			{ 
 				sizeof(BITMAPINFO), 
-				width, height, 1, 
-				bpp, BI_RGB, 
-				width * height * bpp / 8, 
+				width, height, static_cast<WORD>(1),
+				static_cast<WORD>(bpp), BI_RGB, 
+				static_cast<DWORD>(static_cast<unsigned>(width) * height * bpp / 8),
```
Comment (Warepire) [inline wintaser/AVIDumper.cpp]: Cast to something clearer than `unsigned`

File: wintaser/InputCapture.cpp [hunk 2]
```diff
@@ -626,7 +626,7 @@ void InputCapture::ProcessInputs(CurrentInput* currentI, HWND hWnd){
 			continue;
 
 		// Now we build the SingleInput, and check if it's mapped to something.
-		SingleInput siPressed = {SINGLE_INPUT_DI_KEYBOARD, k, ""};
+		SingleInput siPressed = { SINGLE_INPUT_DI_KEYBOARD, static_cast<SHORT>(k), "" };
```
Comment (Warepire) [inline wintaser/InputCapture.cpp:629]: Can ´k´ be a `SHORT` by default?

File: wintaser/MainMenu.cpp [hunk 3]
```diff
@@ -5,24 +5,36 @@
...
 
-#define MENU_L(smenu, pos, flags, id, suffixe, str, disableReason)	do{									\
-/*GetPrivateProfileStringA(language_name[Language], (str), (def), Str_Tmp, 1024, Language_Path);*/	\
-if((disableReason) && ((flags) & (MF_DISABLED|MF_GRAYED))) \
-	sprintf(Str_Tmp, "%s (%s)", str, disableReason); \
-else \
-	strcpy(Str_Tmp, str); \
-strcat(Str_Tmp, (suffixe)); /*AddHotkeySuffix(Str_Tmp, id, suffixe, true);*/							\
-InsertMenu((smenu), (pos), (flags), (id), Str_Tmp); } while(0)
+void MyInsertMenuByID(HMENU smenu, UINT pos, UINT flags, UINT_PTR id, LPCSTR suffixe, LPCSTR str, LPCSTR disableReason)
```
Comment (Warepire) [inline wintaser/MainMenu.cpp]: Does not replace a WinAPI in the DLL, `My` prefix shall not be used.

File: wintaser/MainMenu.cpp [hunk 4]
```diff
@@ -5,24 +5,36 @@
...
+	else
+		strcpy(Str_Tmp, str);
+	strcat(Str_Tmp, suffixe);
+#if 0
+	AddHotkeySuffix(Str_Tmp, id, suffixe, true);
+#endif
+	InsertMenu(smenu, pos, flags, id, Str_Tmp);
+}
 
+__forceinline void MyInsertMenu(HMENU smenu, UINT pos, UINT flags, HMENU menu, LPCSTR suffixe, LPCSTR str, LPCSTR disableReason)
```
Comment (Warepire) [inline wintaser/MainMenu.cpp]: Same here.

File: wintaser/MainMenu.cpp [hunk 5]
```diff
@@ -63,422 +81,439 @@ void Build_Main_Menu(HMENU& MainMenu, HWND hWnd)
...
 
 	Flags = MF_BYPOSITION | MF_STRING;
 	
 	i = 0;
-
-	//MENU_L(Graphics, i++, Flags | (!localTASflags.forceWindowed ? MF_CHECKED : MF_UNCHECKED) | (started ? MF_GRAYED : 0), ID_GRAPHICS_ALLOWFULLSCREEN, "", "Allow &Fullscreen / Display Mode Changes", "can't change while running");
-	//InsertMenu(Graphics, i++, MF_SEPARATOR, NULL, NULL);
-	MENU_L(Graphics, i++, Flags | (!localTASflags.forceSoftware ? MF_CHECKED : MF_UNCHECKED) | (started ? MF_GRAYED : 0), ID_GRAPHICS_FORCESOFTWARE, "", "&Allow Hardware Acceleration", "can't change while running");
-	MENU_L(Graphics, i++, Flags | MF_POPUP | ((localTASflags.forceSoftware||started) ? MF_GRAYED : 0), (UINT)GraphicsMemory, "", "Surface &Memory", started ? "can't change while running" : "hardware acceleration must be enabled");
+#if 0
```
Comment (Warepire) [inline wintaser/MainMenu.cpp]: This `#if 0` can be removed, this one seems to be disabled by mistake somewhere...

**Comment by Warepire** [comment]
Sorry it took so long to look at this. After above comments are addressed, this is a LGTM.

**Comment by Warepire** [comment]
Ping @aqualast

File: wintaser/InputCapture.cpp (hunk 2 above)
Comment (aqualast) [inline wintaser/InputCapture.cpp:629]: I'm unsure about this. It is also used as an array index.

**Comment by aqualast** [comment]
@Warepire I added a few more fixes.

File: wintaser/InputCapture.cpp (hunk 2 above)
Comment (Warepire) [inline wintaser/InputCapture.cpp:629]: It's perfectly ok to index an array with a `SHORT`, as long as the array isn't longer than the max value of a `SHORT`.

**Comment by Warepire** [comment]
Once the `k` variable discussion is resolved, this is LGTM.

File: wintasee/hooks/inputhooks.cpp [hunk 6]
```diff
@@ -1248,7 +1250,7 @@ void ProcessFrameInput()
 			s_lii.dwTime = timeStamp;
 
 			__declspec(noinline) SHORT WINAPI MyGetKeyState(int vKey);
-			DIDEVICEOBJECTDATA keyEvent = {i, MyGetKeyState(i) & 0xFF, timeStamp, inputEventSequenceID++};
+			DIDEVICEOBJECTDATA keyEvent = { i, static_cast<DWORD>(MyGetKeyState(i) & 0xFF), timeStamp, inputEventSequenceID++};
```
Comment (YaLTeR) [inline wintasee/hooks/inputhooks.cpp]: Extra space got here by mistake?

File: wintaser/MainMenu.cpp [hunk 7]
```diff
@@ -5,24 +5,36 @@
...
 
-#define MENU_L(smenu, pos, flags, id, suffixe, str, disableReason)	do{									\
-/*GetPrivateProfileStringA(language_name[Language], (str), (def), Str_Tmp, 1024, Language_Path);*/	\
-if((disableReason) && ((flags) & (MF_DISABLED|MF_GRAYED))) \
-	sprintf(Str_Tmp, "%s (%s)", str, disableReason); \
-else \
-	strcpy(Str_Tmp, str); \
-strcat(Str_Tmp, (suffixe)); /*AddHotkeySuffix(Str_Tmp, id, suffixe, true);*/							\
-InsertMenu((smenu), (pos), (flags), (id), Str_Tmp); } while(0)
+void HelperFuncInsertMenuByID(HMENU smenu, UINT pos, UINT flags, UINT_PTR id, LPCSTR suffixe, LPCSTR str, LPCSTR disableReason)
```
Comment (YaLTeR) [inline wintaser/MainMenu.cpp]: `suffixe` -> `suffix` perhaps? Seems like a typo.

File: wintaser/Movie.cpp [hunk 8]
```diff
@@ -24,6 +24,11 @@ Movie::Movie()
 	author[0] = '\0';
 	commandline[0] = '\0';
 	headerBuilt = false;
+	fps = 0;
+	it = 0;
+	memset(fmd5, 0, 4 * sizeof(unsigned int));
```
Comment (YaLTeR) [inline wintaser/Movie.cpp]: `4 * sizeof(unsigned int)` -> `sizeof(fmd5)`, same below

File: wintaser/InputCapture.cpp (hunk 2 above)
Comment (aqualast) [inline wintaser/InputCapture.cpp:629]: Hold on. Shouldn't ´k´ be an `unsigned char`? It's also passed to ´IsModifier()´ below which takes an `unsigned char` as the parameter.

File: wintaser/MainMenu.cpp (hunk 7 above)
Comment (aqualast) [inline wintaser/MainMenu.cpp]: It's French if you ask me.

File: wintaser/InputCapture.cpp (hunk 2 above)
Comment (Warepire) [inline wintaser/InputCapture.cpp:629]: You're right, keep the casts to `SHORT`, and turn `k` to an `unsigned char`.

**Comment by aqualast** [comment]
Ping @Warepire @YaLTeR

File: wintaser/Movie.cpp [hunk 9]
```diff
@@ -24,6 +24,13 @@ Movie::Movie()
 	author[0] = '\0';
 	commandline[0] = '\0';
 	headerBuilt = false;
+	fps = 0;
+	it = 0;
+	static_assert(!std::is_pointer<decltype(fmd5)>::value, "sizeof(pointer) is wrong!");
```
Comment (Warepire) [inline wintaser/Movie.cpp]: No. Don't `static_assert`

File: wintaser/Movie.cpp [hunk 10]
```diff
@@ -24,6 +24,13 @@ Movie::Movie()
 	author[0] = '\0';
 	commandline[0] = '\0';
 	headerBuilt = false;
+	fps = 0;
+	it = 0;
+	static_assert(!std::is_pointer<decltype(fmd5)>::value, "sizeof(pointer) is wrong!");
+	memset(fmd5, 0, sizeof(fmd5));
+	fsize = 0;
+	static_assert(!std::is_pointer<decltype(desyncDetectionTimerValues)>::value, "sizeof(pointer) is wrong!");
```
Comment (Warepire) [inline wintaser/Movie.cpp]: Same here.

**Comment by Warepire** [comment]
After that I think this is ready to merge.

**Comment by aqualast** [comment]
@Warepire Everything is right now?

**Comment by vadosnaprimer** [comment]
Congrats guys.

**Pull request merged**